hollow/src/main/java/com/netflix/hollow/api/consumer/fs/HollowFilesystemBlobRetriever.java

```diff
@@ -312,7 +339,25 @@ private boolean allRequestedPartsExist(HollowConsumer.Blob.BlobType type, long c
         }
 
         return true;
-    }    
+    }
+
+    private void cleanupOldBlobs() {
+        if (existingBlobMaxAge == null || existingBlobMaxAge.isNegative()) {
+            return;
+        }
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(blobStorePath)) {
+            for (Path blobPath : directoryStream) {
+                long lastModifiedTime = Files.getLastModifiedTime(blobPath).toMillis();
+                long currentTime = System.currentTimeMillis();
+                if ((currentTime - lastModifiedTime) > existingBlobMaxAge.toMillis()) {
+                    Files.delete(blobPath);
+                    LOG.info("Deleted old blob: " + blobPath + " last modified at " + lastModifiedTime);
+                }
+            }
+        } catch (IOException ex) {
+            throw new RuntimeException("Error listing blob files for deletion; path=" + blobStorePath, ex);
```

Comment:
nit: exception msg seems too specific e.g. the exception could also originate in `Files.delete`.. and it could also be useful to log the modified time seen on the file and the configured max age for deletion

Reply:
Updated the exception message to be more general. The file last modify time was already being logged above on deletion, updated to also include the configured max age for deletion.